src/Core/tests/UnitTests/MauiContextTests.cs

```diff
@@ -99,6 +99,23 @@ public void CloneCanOverrideIncludeService()
 			Assert.Same(obj2, second.Services.GetService<TestThing>());
 		}
 
+		[Fact]
+		public void MauiContextSupportsKeyedServices()
```

Comment:
Negative test?

Reply:
All the other tests are the negative. We are not actually implementing a new service provider, but rather have a proxy/wrapper instance that allows for the ability to inject a value for a specific instance of a wrapper. This is needed for Android which has a weird way of doing things, but the wrapper just redirects all calls to the `Inner` instance.